datalake/src/main/java/com/sequenceiq/datalake/cm/ClouderaManagerRangerUtil.java

```diff
@@ -0,0 +1,124 @@
+package com.sequenceiq.datalake.cm;
+
+import com.cloudera.api.swagger.ClustersResourceApi;
+import com.cloudera.api.swagger.RoleCommandsResourceApi;
+import com.cloudera.api.swagger.RolesResourceApi;
+import com.cloudera.api.swagger.client.ApiClient;
+import com.cloudera.api.swagger.client.ApiException;
+import com.cloudera.api.swagger.model.ApiBulkCommandList;
+import com.cloudera.api.swagger.model.ApiClusterList;
+import com.cloudera.api.swagger.model.ApiCommand;
+import com.cloudera.api.swagger.model.ApiConfig;
+import com.cloudera.api.swagger.model.ApiConfigList;
+import com.cloudera.api.swagger.model.ApiRole;
+import com.cloudera.api.swagger.model.ApiRoleNameList;
+import com.google.common.base.Joiner;
+import com.google.common.base.Joiner.MapJoiner;
+import com.google.common.collect.ImmutableSortedMap;
+import com.google.common.collect.Iterables;
+import com.sequenceiq.cloudbreak.cm.client.retry.ClouderaManagerApiFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.retry.RetryException;
+import org.springframework.retry.annotation.Backoff;
+import org.springframework.retry.annotation.Retryable;
+import org.springframework.stereotype.Component;
+
+import javax.inject.Inject;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+@Component
+public class ClouderaManagerRangerUtil {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ClouderaManagerRangerUtil.class);
+
+    private static final String RANGER_SERVICE_NAME = "ranger";
```

Comment:
Maybe you could include the cluster-cm module and move all the CM related code there and use it from Data Lake.

Reply:
Do you mind if we look at this in a follow up task? Moving the logic into cluster-cm is straight forward, but pulling in cluster-cm as dependency seems problematic. I ran into dependency conflicts (hibernate, CachingConfig) as well as it requiring additional dependencies (I think it required bringing an implementation of CloudbreakEventService). The  refactoring required might be beyond the scope of this task. I can file a ticket to see if the CM portion can be easily be moved into cluster-cm and client-cm.